onedal/_device_offload.py

```diff
@@ -153,11 +164,35 @@ def _get_host_inputs(*args, **kwargs):
     return q, hostargs, hostkwargs
 
 
-def _extract_usm_iface(*args, **kwargs):
+def _extract_array_attr(*args, **kwargs):
+    """Extracts USM iface, array namespace and hardware device
+    the array data resides on.
+
+    Returns
+    -------
+    usm_iface : Dict
+        Sycl USM Array (SUA) interface protocol dictionary describing the array.
+    array_api : str
```

Comment:
Are we sure its a string here? Does this clash with what is defined in ```get_namespace``` when ```__array_namespace__``` is called?

Reply:
Outdated. Please follow the discussion at https://github.com/intel/scikit-learn-intelex/pull/1861#discussion_r1746690301